src/Pipeline.cpp

```diff
@@ -582,6 +582,25 @@ void Pipeline::compile_jit(const Target &target_arg) {
     // Clear all cached info in case there is an error.
     contents->invalidate_cache();
 
+#ifdef WITH_SERIALZATION_JIT
+    // TODO: replace file serialization with in-memory serialization
+    std::string filename = generate_function_name() + ".hlpipe";
+    std::map<std::string, Internal::Parameter> external_params;
+    serialize_pipeline(*this, filename, external_params);
+    Pipeline deserialized_pipe = deserialize_pipeline(filename, external_params);
+    std::vector<Function> outputs;
+    for (const Func &f : deserialized_pipe.outputs()) {
+        outputs.push_back(f.function());
+    }
+    // we save the original output functions and requirements
```

Comment:
Grammar nit: I know there are ooooodles of comments in Halide that are just sentence fragments, which is fine, but when a comment is effectively a proper sentence (etc), as this one is, it's good form to give it proper capitalization, punctuation, etc., as it makes future code reading easier on the eyes, eg

`We save the original output functions and requirements, and restore them once all lowering is done, so that reschedule/reorder storage can be properly handled.`